src/codegen/sys/functions.rs

```diff
@@ -183,9 +183,23 @@ fn generate_object_funcs(
         )?;
     }
     if write_get_type {
-        let configured_functions = obj.functions.matched(&glib_get_type);
-        generate_cfg_configure(w, &configured_functions, false)?;
-        writeln!(w, "    pub fn {}() -> GType;", glib_get_type)?;
+        use crate::case::CaseExt;
+
+        let type_name = c_type.to_snake();
+        assert!(glib_get_type.starts_with(&type_name));
+        let function_name = &glib_get_type[(type_name.len() + 1)..];
```

Comment:
Do we have a better way for doing the above 3 steps?

Reply:
Could go via the symbol-prefixes of the namespace and then the symbol-prefix of the type... doing so now, needs more parsing :)